mlir/utils/jenkins/Jenkinsfile.release

```diff
@@ -0,0 +1,96 @@
+// ON CHANGING THESE, ALSO CHANGE Jenkinsfile.downstream
+void buildProject(String target, String cmakeOpts) {
+    cmakeBuild generator: 'Ninja',\
+        buildDir: 'build',\
+        buildType: 'RelWithDebInfo',\
+        installation: 'InSearchPath',\
+        steps: [[args: target]],\
+        cmakeArgs: "-DMLIR_MIOPEN_DRIVER_ENABLED=1 $cmakeOpts"
+}
+
+void showEnv() {
+    echo "$env.NODE_NAME"
+    sh 'cat /etc/os-release'
+    sh 'ulimit -a'
+    // Ignore rocm-smi failures in ixt-sjc2-05
+    sh '/opt/rocm/bin/rocm-smi || true'
+    sh '/opt/rocm/bin/rocm_agent_enumerator'
+    sh 'id'
+}
+
+String dockerArgs() {
+    return "--device=/dev/kfd --device=/dev/dri --group-add video -v /etc/passwd:/etc/passwd:ro -v /etc/group:/etc/group:ro"
+}
+
+String dockerImage() {
+    return 'rocm/mlir:rocm4.3-latest'
+}
+
+pipeline {
+    agent none
+    parameters {
+        // Below should be set statically by Jenkins job
+        string(name: 'releaseBranch', defaultValue: '', description: 'release branch name')
+    }
+    stages {
+        stage("Set System Property") {
+            steps {
+                script {
+                     System.setProperty("org.jenkinsci.plugins.durabletask.BourneShellScript.HEARTBEAT_CHECK_INTERVAL", "86400");
+                }
+            }
+        }
+        stage("Store a Release Build") {
+            when {
+                beforeAgent true;
+                not {equals expected: '', actual: params.releaseBranch}
+            }
+            agent {
+                docker {
+                    image dockerImage()
+                    args dockerArgs()
+                    label "rocm"
+                    alwaysPull true
+                }
+            }
+            stages{
+                stage ('Build and Archive a Release Branch') {
+                    environment {
+                        HOME="${WORKSPACE}"
+                        branch=sh(returnStdout:true, script:"""#!/bin/bash
+                            temp=$params.releaseBranch
+                            release=\${temp////-}
+                            echo \$release""").trim()
+                    }
+                    steps {
+                        showEnv()
+                        sh "git checkout ${params.releaseBranch}"
+                        sh "tar czf mlir-source-${branch}.tar.gz \$(ls)"
```

Comment:
Would it be possible to have the tarball unpack into a directory. That is, instead of having the structure be `external/...`, `mlir/...`/, ..., we'd have `miir-${branch}/external/...` `miir-source-${branch}/mlir/...` and so on?

Tarballs that surprise-expand into the current directory are rather annoying, mostly because they're surprising.

Reply:
I think the easiest way is temporarily move everything under a directory then create a tarball.

Reply:
Looks like there's an option to tar for doing things like this https://serverfault.com/questions/86478/tar-how-to-create-a-tar-file-with-arbitrary-leading-directories-w-o-cding-to - I'd look at the second answer with `--transform`

Having tested out, something like this'll do 
```
tar --transform 's#^#a-1.0/#' -cvzf a-1.0.tar.gz *
```

also noting that * is probably better-behaved than $(ls)